src/pages/Stream/providers/LogsProvider.tsx

```diff
@@ -688,7 +688,8 @@ const setCleanStoreForStreamChange = (store: LogsStore) => {
 	const { tableOpts, timeRange, alerts } = store;
 	const { interval, type } = timeRange;
 	const duration = _.find(FIXED_DURATIONS, (duration) => duration.milliseconds === timeRange.interval);
-	const updatedTimeRange = interval && type === 'fixed' ? { timeRange: getDefaultTimeRange(duration) } : { timeRange };
+	const updatedTimeRange = interval && type === 'fixed' ? { ...getDefaultTimeRange(duration) } : { ...timeRange };
+	const isTimeRangeSet = _.isEqual(timeRange, updatedTimeRange);
```

Comment:
why do we need the check here ?

Reply:
So currently the timeRange update happens on Stream (index.ts) via `setInitialTimeRange` and again on mount of useLogsFetcher via `setCleanStoreForStreamChange`. To prevent multiple timeRange updates getting triggered we need some kind of check.

Reply:
Thats right. We have to fix the trigger. Not the implementation.

Reply:
We need both the triggers right as one is setting the timeRange and the other is also setting some tableOptions.

Reply:
> We need both the triggers right as one is setting the timeRange and the other is also setting some tableOptions.

My understanding was wrong. Have addressed this.